src/transformers/models/longformer/modeling_longformer.py

```diff
@@ -1350,14 +1347,15 @@ def custom_forward(*inputs):
             all_hidden_states = all_hidden_states + (hidden_states,)
 
         # undo padding
+        # this path should be recorded in the ONNX export, it is fine with padding_len == 0 as well
         if padding_len > 0:
```

Comment:
Any reason not to remove this condition `if padding_len > 0:`?

Reply:
I think we can remove this `if` statement, and keep your change `hidden_states.shape[1] - padding_len` below.
This might be clear. (unless there is some optimization consideration, i.e. we don't want to enter this block to save some running time)

Reply:
Yes I left it thinking the case `padding_len == 0` would be faster if we don't go in this if code. But I'll remove the condition, I agree it's not clear currently.